libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1778,9 +1778,11 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 			}
 			if (resultType->category() == Type::Category::Address)
 			{
-				bool payable = true;
+				bool payable = false;
 				if (auto const* contractType = dynamic_cast<ContractType const*>(argType.get()))
 					payable = contractType->isPayable();
+				else if (argType->category() == Type::Category::RationalNumber)
```

Comment:
I have the impression that we need more cases here, but there were no test failures.